tools/online_editor/src/worker/lsp_worker.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+import {
+  createConnection,
+  BrowserMessageReader,
+  BrowserMessageWriter,
+} from "vscode-languageserver/browser";
+import {
+  ExecuteCommandParams,
+  InitializeParams,
+  InitializeResult,
+} from "vscode-languageserver";
+
+import slint_init, * as slint_lsp from "@lsp/slint_lsp_wasm.js";
+
+console.log("*** LSP WEBWORKER STARTUP INITIATED! ***");
+
+slint_init().then((_) => {
+  const reader = new BrowserMessageReader(self);
+  const writer = new BrowserMessageWriter(self);
+
+  let the_lsp: slint_lsp.SlintServer;
+
+  const connection = createConnection(reader, writer);
+
+  function send_notification(method: string, params: unknown): boolean {
+    connection.sendNotification(method, params);
+    return true;
+  }
+
+  async function load_file(path: string): Promise<Uint8Array> {
+    return await connection.sendRequest("slint/load_file", path);
+  }
+
+  connection.onInitialize((params: InitializeParams): InitializeResult => {
+    the_lsp = slint_lsp.create(params, send_notification, load_file);
+    return { capabilities: the_lsp.capabilities() };
+  });
+
+  connection.onRequest(async (method, params, token) => {
+    if (
+      method === "workspace/executeCommand" &&
+      (params as ExecuteCommandParams).command === "showPreview"
```

Comment:
I wonder if it makes sense, in the online code editor to have this feature.

We should indeed probably disable the code lense and preview action from the lsp server.
This code is what happens when the client execute a code lense that is meant to show the preview, and it then re-foward it back to the client so that it can update the preview.

We'll have to think if this is something we want to have.  But as for now, the client won't understand the "slint/showPreview" anyhow.

Reply:
Removed!